core/src/main/scala/org/apache/spark/sql/sources/SnappyOptimizations.scala

```diff
@@ -408,7 +409,13 @@ case class ResolveIndex(implicit val snappySession: SnappySession) extends Rule[
 
   override def apply(plan: LogicalPlan): LogicalPlan = {
     val joinOrderHints = JoinOrderStrategy.getJoinOrderHints
-
+    val enabled: Boolean = io.snappydata.Property.EnableExperimentalFeatures.
+        configEntry.getConf[Boolean](snappySession.snappyContext.conf) ||
+        java.lang.Boolean.getBoolean(io.snappydata.Property.EnableExperimentalFeatures
+            .configEntry.key)
```

Comment:
I think we should avoid this explicit system property otherwise it will become a general practise which is not a good idea. I think better option will be to allow alternate "spark." (like done in Property.ValueAlt but not there for SQLVal) and let SparkConf handle those properties for both embedded and split modes. Let me see adding alternative values like done in ValueAlt for SQLVal too (or if you can do it then will be good since I have lots of other stuff to complete urgently).